lib/tests/formulae_detect.rb

```diff
@@ -75,7 +75,13 @@ def detect_formulae!(args:)
           else
             test git, "-C", repository, "fetch", "origin", "+#{github_ref}" unless tap.official?
             origin_ref = "origin/#{github_ref.gsub(%r{^refs/heads/}, "")}"
-            diff_end_sha1 = diff_start_sha1 = github_sha
+            diff_end_sha1 = github_sha
+            diff_start_sha1 = if (event_name = ENV.fetch("GITHUB_EVENT_NAME", nil)).present? &&
+                                 event_name == "merge_group"
```

Comment:
I think this is different enough to be worth splitting into a separate `elsif` above.

Also:

```suggestion
            diff_start_sha1 = if (ENV.fetch("GITHUB_EVENT_NAME", nil) == "merge_group"
```

or similar seems sufficient